File: Leetcode/Leetcode_557.java
// 反转字符串中的单词 III
class Solution {
    public String reverseWords(String s) {
        StringBuffer res = new StringBuffer();;
        int i = 0;
        int n = s.length();
        while (i < n) {
            int start = i;
            while (i < n && s.charAt(i) != ' ') {
                ++i;
            }
            for (int j = i - 1; j >= start; --j) {
                res.append(s.charAt(j));
            }
            while (i < n && s.charAt(i) == ' ') {
                res.append(' ');
                ++i;
            }
        }
        return res.toString();
    }
}

File: Leetcode/Leetcode_557.cpp
// 反转字符串中的单词 III
class Solution {
public:
    string reverseWords(string s) {
        int n = s.size();
        int begin = 0, end;
        for(int i = 0; i < n + 1; ++i) {
            if(s[i] == ' ' || s[i] == '\0') {
                for(end = i - 1; begin < end; begin++, end--) {
                    swap(s[begin], s[end]);
                }
                begin = i + 1;
            }
        }
        return s;
    }
};